### PR TITLE
Fix ebnf errors

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Fix ebnf errors

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-with.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ CommonTableExpr ::=
 
 ```ebnf+diagram
 IdentListWithParenOpt ::=
-|       '(' IdentList ')'
+empty | '(' IdentList ')'
```
Comment (wjhuang2016) [inline sql-statements/sql-statement-with.md:39]:
```suggestion
( '(' IdentList ')' )?
```

**Review by wjhuang2016 — APPROVED** [review]

File: sql-statements/sql-statement-show-create-placement-policy.md [hunk 2]
```diff
@@ -14,7 +14,8 @@ summary: TiDB 数据库中 SHOW CREATE PLACEMENT POLICY 的使用概况。
 ## 语法图
 
 ```ebnf+diagram
-"SHOW" "CREATE" "PLACEMENT" "POLICY" PolicyName
+ShowCreatePlacementPolicyStmt ::=
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply sql-statements/sql-statement-show-create-placement-policy.md:17]: 这里经 SQL 的研发确认没问题，无需修改

**Pull request merged**